### ISSUE TITLE
FormGroup not cleared

### ISSUE THREAD

**Issue by Nielsb85** [ngx-formly/ngx-formly#3286]
**Description**
I have a dynamic form based on routing. based on the queryParam a step within the configuration is loaded in to formly.
This is working great. However the formGroup does not get cleared properly when going to the next or previous step.
This results in the form trying to validate a step which has required fields and my form will not pass the validation.

**Minimal Reproduction**

https://angular-ivy-2bnapy.stackblitz.io

Cant get material to work properly. but when you enter a value on step 0 and press next, it will navigate to step 1 but the formGroup (as you can see in the printed json) still holds the formControls of the previous step even though the config only loads the next step.

I now manually check the differences and remove the controls from the form myself.

**Your Environment**
Angular 13.3.2
 "@ngx-formly/core": "6.0.0-next.9",
"@ngx-formly/material": "6.0.0-next.9",

Also used Formly version 5 (latest). but same result

**Additional context**

When I create a new FormGroup myself on each step, my component goes in to the ngOnDestroy so that is not an option.

**Comment by MaxKless** [comment]
Have you figured out a solution for this problem? Resetting the form group seems like the most obvious solution.
I'm not 100% sure what the intended behavior is here, and the source code (with the `FormlyFieldConfigCache`) is not super easy to read.

You can definitely check out the place where this happens and see if you can learn from that:
https://github.com/ngx-formly/ngx-formly/blob/main/src/core/src/lib/components/formly.form.ts#L85
The `clearControl` method and `FormlyFieldConfigCache` are involved in the handling here. 

Let me know if that helps in any way or if you have an idea how formly could fix it.

**Comment by Nielsb85** [comment]
resetting the formgroup will result in angular calling the ngDestroy on the component where the formgroup was set. (I dont know why)

I do have a workaround within the app itself.
I would say that the intended behaviour would be that the formgroup respects the configuration and instead of pushing new values it clears the old one and generates a new formgroup based on the current configuration. But that is my guess.

I was indeed checking the source and I agree it was not easy to read so I am not sure. 

```
export function clearControl(form: FormlyFieldConfigCache['formControl']) {
  delete form?._fields;
  form.setValidators(null);
  form.setAsyncValidators(null);
  if (form instanceof FormGroup || form instanceof FormArray) {
    Object.values(form.controls).forEach((c) => clearControl(c));
  }
}
```

If you look at the clearControl function it deletes private properties from the formControl. Instead it should loop over the formGroup controls and call the clearControl method. as I think under the hood that will invoke other methods that makes sure the formgroup is properly cleared.

**Comment by kenisteward** [comment]
I agree with @maxkless here. If the idea Is that you have different forms
per query param the a new formly form should be instantiated.

The stepper setup I believe was originally intended for a single form with
multiple steps.

**Comment by aitboudad** [comment]
> You can definitely check out the place where this happens and see if you can learn from that:
https://github.com/ngx-formly/ngx-formly/blob/main/src/core/src/lib/components/formly.form.ts#L85
The `clearControl` method and `FormlyFieldConfigCache` are involved in the handling here. 

That concern only validations but I do agree that we should clear controls as well but not a straight task as we need to extract the fields to remove from by comparing the prev/new fields, the easy solution is to clear `FormGroup` by you side.

Can you try the following solution: https://stackblitz.com/edit/angular-ivy-yspupc?file=src/app/app.component.html
which rely on `modelChange` output.

**Comment by MaxKless** [comment]
@aitboudad ok makes sense, removing the fields that disappear after new fields are loaded seems like the right solution - but somewhat complicated.

I would like to try and contribute but I have to admit I haven't really understood the whole `FormlyFieldConfigCache` yet (as you can see by my comment above) and I assume I would have to work with that. 

Any way you could give me some pointers as to how you would go about this change?

**Comment by aitboudad** [comment]
> I would like to try and contribute but I have to admit I haven't really understood the whole `FormlyFieldConfigCache` yet (as you can see by my comment above) and I assume I would have to work with that. 

During the build step we add some additional information to the field such as `_validators`, `formControl`, 'model' and more. Some of these properties are not part of our public API (`FormlyFieldConfig`) that's where `FormlyFieldConfigCache` is used which represent the `field` internal API.

> Any way you could give me some pointers as to how you would go about this change? 

Not a priority for now but anyway I would say you can use [`clearControl`](https://github.com/ngx-formly/ngx-formly/blob/main/src/core/src/lib/components/formly.form.ts#L87), pass the following arguments `fields`  (`previousValue` and `currentValue`) and `forms` and use [unregisterControl](https://github.com/ngx-formly/ngx-formly/blob/main/src/core/src/lib/extensions/field-form/utils.ts#L6)
